src/Scheduler.scala

```diff
@@ -0,0 +1,20 @@
+import java.util.concurrent.{Future, PriorityBlockingQueue}
+
+import scala.concurrent.ExecutionContext
+
+trait Scheduler {
+  /**
+    * should include the queue as a property like class?
```

Comment:
In this case the trait servers only as an interface, and the queue is an implementation detail. This may not always make sense and this could just be the class definition if there were not multiple implementations.